src/mca/psec/base/psec_base_frame.c

```diff
@@ -43,24 +43,31 @@
 #include "src/mca/psec/base/static-components.h"
 
 /* Instantiate the global vars */
-pmix_psec_globals_t pmix_psec_globals = {0};
+pmix_psec_globals_t pmix_psec_globals = {{{0}}};
 
 static pmix_status_t pmix_psec_close(void)
 {
-  pmix_psec_base_active_module_t *active;
+  pmix_psec_base_active_module_t *active, *prev;
+  pmix_list_item_t *item, *prev_item;
 
     if (!pmix_psec_globals.initialized) {
         return PMIX_SUCCESS;
     }
     pmix_psec_globals.initialized = false;
 
-    PMIX_LIST_FOREACH(active, &pmix_psec_globals.actives, pmix_psec_base_active_module_t) {
+    PMIX_LIST_FOREACH_SAFE(active, prev, &pmix_psec_globals.actives, pmix_psec_base_active_module_t) {
+      pmix_list_remove_item(&pmix_psec_globals.actives, &active->super);
       if (NULL != active->component->finalize) {
         active->component->finalize();
       }
+      PMIX_RELEASE(active);
     }
     PMIX_DESTRUCT(&pmix_psec_globals.actives);
 
+    PMIX_LIST_FOREACH_SAFE(item, prev_item, &pmix_psec_base_framework.framework_components, pmix_list_item_t) {
```

Comment:
Are you saying that framework_close doesn't destroy this list for us? That would seem like a bug in the MCA base, yes?